src/led_strip.py

```diff
@@ -2,15 +2,58 @@
 from time import sleep
 import uasyncio as asyncio
 
+# :TODO: Unsure on naming of this mixin/class. Also once that's settled, move to another file.
+class LedGestures:
+    # set to a new color
+    async def until_color_changed(self, color): 
+        self.pixels.fill(color)
+        self.pixels.write()
+
+    # Turn off the lights
+    async def until_off(self): 
+        await self.until_color_changed((0,0,0,0))
+
+    # Reset to the configured color
+    async def until_reset(self):
+        await self.until_color_changed(self.color)
+
+    # This is gross. I should learn python better!
+    # This currently works but allocates way too much memory and is pretty terrible.
+    # This should be able to be cleaned up to store less and loop less.
+    # Probably replace steps with time in seconds when this is rewritten. 
+    async def until_faded_to(self, color, steps = 50):
```

Comment:
I had an idea to abstract out the setting of the colors from the actual `pixels.write()`. I was thinking we might want to have a simple Task that's running in parallel that just calls `write()` at the set frame rate, and then sets an Event that any tasks that are doing an animation can wait on to know when it's time to 'draw' the next set of colors.
Here's an example: https://gist.github.com/cherrydev/cd12a7f0da2e44b477ff656264f13e06
This still looks like a good starting place, though.